### ISSUE TITLE
Apps on external app stores

### ISSUE THREAD

**Issue by LEDfan** [nextcloud/appstore#95]
Hi, I hope it isn't asked before :)

There are quite some applications which work together with Nextcloud but aren't Nextcloud apps. Some examples: 
- https://apps.owncloud.com/content/show.php/QOwnNotes?content=168497
- https://github.com/aykit/MyOwnNotes
- https://github.com/owncloud/news-iOS-App
- https://github.com/owncloud/News-Android-App
- https://github.com/owncloud/notes-iOS-App

To boost the Nextcloud ecosystem I believe it's very important to allow this sort of apps in the appstore. Of course clearly state the platform, and the external link to the app store. (Be it OBS, Google play etc)

What do you think?

P.s. this can be done in later releases ;)

**Comment by BernhardPosselt** [comment]
I dont think they should be present in the app store since they don't fit in (analogy: The Nextcloud News app is also not available on the Android and iOS Store since you can't install them there). 

Rather the app itself should provide a list of known apps which can be integrated, check out https://github.com/nextcloud/news#sync-clients

**Comment by BernhardPosselt** [comment]
PS: App descriptions support markdown so you can add them there. The only question to keep this ticket open that I see here is if we need explicit markup in the info.xml, e.g. a \<sync-client> tag, however this is probably way to specific and might not fit the usecases people have.

Closing hence :)

**Comment by BernhardPosselt** [comment]
Another thing: Apps like QOwnNotes regularely pushed updates on the app store to notify people about new releases. I don't think this channel fits the app. People that use that app should get these updates from their repo in the same way that users of the android an iOS apps should be notified about new version on their app store.

So basically allowing these kinds of apps on the app store make maintainers think that it's a good idea to also push updates which is neither good for the dev nor the app's users.

**Comment by LEDfan** [comment]
> So basically allowing these kinds of apps on the app store make maintainers think that it's a good idea to also push updates which is neither good for the dev nor the app's users.

It would only be used to inform users of the existence of the app. Not to push updates since that's something for specific app stores. 

> Rather the app itself should provide a list of known apps which can be integrated, check out https://github.com/nextcloud/news#sync-clients 

That's basically where I'm asking for :smiley:, but in a more structured way.

**Comment by BernhardPosselt** [comment]
@LEDfan what kind of structure are you thinking of :)

**Comment by BernhardPosselt** [comment]
Maybe a tag like: \<related-app is-foss="true" url="https://link-to-app.com">Link name</related-app> in your info.xml?

**Comment by LEDfan** [comment]
I think that would be indeed enough information. Then a section with those related apps would be good, a bit like the reviews section in sandstorm: e.g. https://apps.sandstorm.io/app/vfnwptfn02ty21w715snyyczw0nqxkv3jvawcah10c6z7hj1hnu0

**Comment by janibonnevier** [comment]
If it is very common for apps to have such related external apps, then I'm all for special tag. Apart from enabling pretty presentation, the data could be used for stats and maybe as search criteria in an advanced app search.

Otherwise, I think the apps that do have related apps can make do with the markdown in descriptions.

**Comment by BernhardPosselt** [comment]
It's not that common (maybe 10 of 250 apps have a related app) but the most popular apps usually have related apps

Like you've mentioned the linked apps can be useful (e.g I want all notes apps that have a desktop client) but that depends on the quality that we use for defining the data structure and I'm uncertain what the best solution would look like. 

Maybe this is a feature that becomes relevant at some point but I'm unsure if we can define a good solution ATM. I want to avoid api breaks so maybe we should postpone the decision

**Comment by olantrust** [comment]
Nextcloud users will look at the appstore to determine what apps are associated with Nextcloud, even before they choose to use Nextcloud. In a 'normal users' brain they will assume that what's on the appstore is what's available to use. 

Most Users will not differentiate between our appstore, and for example, Google Play appstore, they are too busy with their own lives and expect things to just 'work' whichever way they approach them. The same applies for people 'checking out' if Nextcloud would suit them. They won't bother checking each repo for each of their user types (Windows Tablet, Windows Phone, Android, Ubuntu, Mac...)
 
Let's make it easy for them, by having a 'Computer and Phone Apps' section in our appstore to display and list any associated external apps (mobile, desktop, tablet). They will then get a link to install them via the appropriate repo, and get updates from that repo (or exceptionally, a download link).

Earlier in this convo it's stated:
"I dont think they should be present in the app store since they don't fit in (analogy: The Nextcloud News app is also not available on the Android and iOS Store since you can't install them there)."

Strongly disagree. These examples are not analogous. Nextcloud appstore in the eyes of the user is the place for anything to do with Nextcloud, not a repo. This is about usability. 
There is no harm in providing links to external apps in their repos. There is harm in not providing it. 
And this is the place to provide it since this is where people will look for 'apps'.

**Comment by BernhardPosselt** [comment]
@olantrust users want to run and install Nextcloud apps when going to the Nextcloud store. Only after that they will want to check which apps integrate with the ones that they installed. Placing links to those apps on the app's detail page in whatever form is the best solution since this correlates the app with compatible apps. It also enables users to search for "news android" which will display the News app containing the links.

Anyways we will definitely not let developers upload your c++/java/whatever code on the App Store since its almost always the worst possible ways for users to install your software. Your argument is more about discoverability: Can a user easily find out which apps integrate with a particular app. We are trying to find out what's the best way to do this :)

**Comment by olantrust** [comment]
@BernhardPosselt Partially agree. Yes: that downloading the app from the store would be a bad idea. Links to easily install via the repo would be the appropriate way.
But disagree about only linking from existing apps. Users also want to find an external app that works for their system and device, then see what Nextcloud apps I need for it.
As Users or potential users we want an easy place to see with our devices what features we can use.

What is the issue with also having a section for External Apps - if it provides links to repos, and an overview making it much quicker and easier for users and prospective users to see what's currently available for each device and OS?
That makes it easy. Hiding that info only in links inside the from a whole gamut of internal apps makes it difficult.

**Comment by BernhardPosselt** [comment]
So in other words: we want users to be able to search for other operating systems and filter Nextcloud apps that have 3rdparty apps that integrate with it?

**Comment by olantrust** [comment]
Hmm, to have a section in the Nextcloud appstore which shows Nextcloud companion (external) apps.
User can filter by system (Windows, iOS, Mac, Android, Chrome, Linux, Other).
Each of those app entries can list: a) external repos where they can found, b) which Nextcloud apps they work with.

**Comment by BernhardPosselt** [comment]
To achieve something like this an app author could add something like mentioned above to his info.xml:
```xml
<related-app>
    <name>News Android App</name>
    <license>proprietary</license>
    <os>Android</os>
    <url>https://play.google.com/store/apps/details?id=de.luhmer.owncloudnewsreader</url>
</related-app>
```

App devs that want their apps listed as compatible app can simply make a pull request and the next time a release is updated it will appear on the app detail page. 

After that we can also think of a section with external apps which basically consists of a list of links :)

The main thing I want to avoid is that external app devs will have to maintain yet another store page.

**Comment by olantrust** [comment]
@BernhardPosselt Great plan. Agree that any external app section should be sustainable and not time-consuming.

**Comment by korelstar** [comment]
Nice idea. One suggestion: I would name the attribute for the system on which the app runs on `platform` (instead of `os`). Sometimes, we are not talking about an operating system, but about a platform, e.g. Java, ...

**Comment by e-alfred** [comment]
Are links to external apps a good idea? If they are Nextcloud apps (not companion apps for Nextcloud apps), why not put them in the store? Most of those not in the store are either abandoned or not compatible with (recent) Nextcloud releases. The Nextcloud app store shouldn't be littered with abandoned apps like the old Owncloud app store. It makes handling and use of the store quite difficult and cluttered.

Links to companion apps in the description would be great to make people aware that these exist on the other hand.

**Comment by sunjam** [comment]
I fully support an external apps section of the Apps Store.  It could also be described as "Community Submissions."

**Comment by mario** [comment]
@jospoortvliet do you think our new Integrations magic should be used to promote 3rd party mobile clients as well, like requested here?

**Comment by mario** [comment]
I guess so, so I'm gonna close this as fulfilled - done :)

Please use the new integrations submission process.

**Comment by korelstar** [comment]
@ma12-co can you give a pointer to more information about that, please?

**Comment by mario** [comment]
@korelstar he can't, but I can :P

So once you log in to the app store, you'll see a new menu that says "Integration developer". Thanks to that you are now able to submit integrations, like for example the Deck Android app.

It needs to pass a review process, but it's pretty simple and mostly in place to get rid of spam submissions.

**Comment by sunjam** [comment]
> So once you log in to the app store, you'll see a new menu that says "Integration developer". Thanks to that you are now able to submit integrations, like for example the Deck Android app.
> 

Thanks, We should add this information to Nextcloud documentation.

**Comment by mario** [comment]
@sunjam the need for documentation is tracked here: https://github.com/nextcloud/appstore/issues/670

I'm just very busy :(

**Comment by korelstar** [comment]
> @korelstar he can't, but I can :P

Sorry, I mixed up the nick names when GitHub made suggestions ... Anyway, thanks for the info!

**Comment by e-alfred** [comment]
@mario I registered the first integration (ocSMS Android app), where will it show up?

**Comment by mario** [comment]
@e-alfred in the integrations category once approved.

**Comment by e-alfred** [comment]
But wouldn't it be better to have a different category like "External Integrations" as to not mix it with normal Nextcloud apps in the existing category? Otherwise it is a bit cluttered I think.

**Comment by mario** [comment]
@e-alfred the right address for this question is @jospoortvliet

**Comment by sunjam** [comment]
I suggest a colorful tag, perhaps in addition to a new category.

**Comment by mario** [comment]
@e-alfred why not put up a screenshot to your entry? :) Go to account/integrations and add one!

**Comment by stefan-niedermann** [comment]
Might be also interesting for @phedlund (Notes iOS app): You can now submit your App at the [Nextcloud app store](https://apps.nextcloud.com), so it is easier to find for the users.

This is how it will look like:

![grafik](https://user-images.githubusercontent.com/4741199/80734694-d440ac80-8b0f-11ea-9d61-2f8b02722e64.png)

**Comment by phedlund** [comment]
Interesting. I'll look into adding my apps.

**Comment by Spartachetto** [comment]
I have seen that @eneiluj apps are on gitlab. This mention is an attempt to spread the news...

**Comment by e-alfred** [comment]
@mario Fixed it some time ago, but the browser extensions I added aren't showing up in the App Store yet. :)
